client/src/components/base-editor/base-editor.tsx

```diff
@@ -24,24 +27,94 @@ const BaseEditor = forwardRef(
     ref: React.Ref<MonacoEditor>,
   ) => {
     const { textEditorSettings } = useContext(SettingsContext)
+    const { customBinding, theme } = textEditorSettings
+
+    const boundEditor = useRef<any>(undefined)
+    const editor = useRef<monaco.editor.IStandaloneCodeEditor | undefined>(
+      undefined,
+    )
+    const actionsContainer = useRef<HTMLDivElement | null>(null)
+
+    useEffect(() => {
```

Comment:
This code can be simplified as 

```typescript
useEffect(() => {
  // if there is any previous editorPlugin initialized, it can be safely disposed
  editorPlugin.current?.dispose()

  if (!editor.current) {
    return
  }

  switch(customBinding) {
    case Keybindings.VIM: {
      if (actionsContainer.current) {
        editorPlugin.current = initVimMode(editor.current, actionsContainer.current)
      }
      break
    }
    case Keybindings.EMACS: {
      editorPlugin.current = initEmacsMode(editor.current)
    }
  }
}, [customBinding])
```